src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -234,11 +266,6 @@ protected AnalyzerDriver(ImmutableArray<DiagnosticAnalyzer> analyzers, AnalyzerM
 
             this.CompilationEventQueue = new AsyncQueue<CompilationEvent>();
             this.DiagnosticQueue = new AsyncQueue<Diagnostic>();
-            _queueRegistration = cancellationToken.Register(() =>
```

Comment:
Hmm this is interesting. This change would essentially make the analysis non-cancellable, which I am not sure is desirable for command line builds. We want to allow users to explicitly cancel out long running analysis through cancelling builds.
Probably we should instead retain this code, but change all the current Enqueue invocations on both the queues to TryEnqueue?